arch/common/CMakeLists.txt

```diff
@@ -12,3 +12,9 @@ zephyr_sources_ifdef(
   CONFIG_EXECUTION_BENCHMARKING
   timing_info_bench.c
   )
+
+
+zephyr_linker_sources_ifdef(CONFIG_ARCH_HAS_RAMFUNC_SUPPORT
```

Comment:
Trying to understand consequences and possibilities this offers.
So, suppose a soc requires custom ramfunc support, I'd just have to:
-not to select ARCH_HAS_RAMFUNC_SUPPORT
-add the equivalent in of this code in soc/x/y/CMakeLists.txt
-provide the ad hoc ramfunc.ld where it makes sense
Is that correct understanding?

Reply:
Yes, that's the gist of it. An alternate solution would be to ifdef in the existing ramfunc.ld.

Reply:
Well, this is more than welcome, then! Thanks.
Let me look have a look to other parts and see if it solves few other pending points.